### DISCUSSION TITLE
Some undefined parameters from the filecoin spec

### DISCUSSION THREAD

**Discussion by dwarkeshsp** [filecoin-project/community#248] [Q&A]
Hey all,

As a self-teaching technique, I'm trying to reimplement the filecoin proof of replication using the spec. There's a few parameters that came up which weren't clearly defined. I'm wondering if any of you can help me understand where they come from and what they mean.

Here's the section of the Filecoin spec they come from: https://spec.filecoin.io/#section-algorithms.sdr.replication

Here are the terms: ProverID, SectorID, PorepID, R_replica_id (different from ReplicaID)

If you have some context on any one of them, I would greatly appreciate a clarifying comment. Thanks!

**Comment by longfeiWan9** [comment]
After reading the spec you pointed to, I understand as :

- ProverID: Miner who wants to generate replicateID
- SectorID: This is sector associates with replica - R
- PorepID: Proof of replication ID
-  R_replica_id (different from ReplicaID): Unique random value for each ReplicaID
